File: shared/serialization/src/commonTest/kotlin/kmp/benchmarks/serialization/kotlinxserialization/JsonData.kt
package kmp.benchmarks.serialization.kotlinxserialization

import kmp.benchmarks.serialization.JsonStrings

object JsonData {
    val largeList = KotlinxSerialization.decodeLargeListFromString(JsonStrings.Minimised.largeList)
    val largeListCompact = KotlinxSerialization.decodeLargeListCompactFromString(JsonStrings.Minimised.largeList)
    val macosReleases = KotlinxSerialization.decodeMacOsReleasesFromString(JsonStrings.Minimised.macosReleases)
    val polymorphicGeo = KotlinxSerialization.decodePolymorphicGeoFromString(JsonStrings.Minimised.polymorphicGeo)
    val polymorphicHtml = KotlinxSerialization.decodePolymorphicHtmlFromString(JsonStrings.Minimised.polymorphicHtml)
    val userProfile = KotlinxSerialization.decodeUserProfileFromString(JsonStrings.Minimised.userProfile)
}

File: app-ios/multiplatformTests/serialization/JsonData.swift
//
//  JsonData.swift
//  multiplatformTests
//
//  Created by Sebastian Owodzin on 25/01/2024.
//  Copyright © 2024 kmp. All rights reserved.
//

import BenchmarksShared

public class JsonData {
    public static var largeList: [GithubPushNormal] {
        return try! KotlinxSerialization().decodeLargeListFromString(jsonString: JsonStrings.Minimised.largeList)
    }

    public static var largeListCompact: [GithubPushCompact] {
        return try! KotlinxSerialization().decodeLargeListCompactFromString(jsonString: JsonStrings.Minimised.largeList)
    }
    
    public static var macosReleases: MacOsReleases {
        return try! KotlinxSerialization().decodeMacOsReleasesFromString(jsonString: JsonStrings.Minimised.macosReleases)
    }
    
    public static var polymorphicGeo: [GeoJSONObject] {
        return try! KotlinxSerialization().decodePolymorphicGeoFromString(jsonString: JsonStrings.Minimised.polymorphicGeo)
    }
    
    public static var polymorphicHtml: [HtmlChunk] {
        return try! KotlinxSerialization().decodePolymorphicHtmlFromString(jsonString: JsonStrings.Minimised.polymorphicHtml)
    }
    
    public static var userProfile: UserProfile {
        return try! KotlinxSerialization().decodeUserProfileFromString(jsonString: JsonStrings.Minimised.userProfile)
    }
}
